### PR TITLE
[MediaTool] Add audio channel mapping APIs for PCM format

### PR DESCRIPTION
**Pull request by hsgwon**
### Description of Change ###

Add audio channel mapping APIs for PCM format

### API Changes ###

 - TCSACR-275

### CODE REVIEW

File: src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs [hunk 1]
```diff
@@ -180,23 +324,38 @@ internal override void AsNativeHandle(IntPtr handle)
...
             MultimediaDebug.AssertNoError(ret);
+
+            if (AudioChannelMap != null)
+            {
+                ret = Native.SetAudioChannelMask(handle, GetAudioChannelMask(handle));
+                MultimediaDebug.AssertNoError(ret);
+            }
+        }
+
+        private ulong GetAudioChannelMask(IntPtr handle)
```
Comment (myroot) [inline src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs]:
```suggestion
        private static ulong GetAudioChannelMask(IntPtr handle, IList<MediaFormatAudioChannelPosition> audioChannelMap )
```
How about change to static?

Comment (hsgwon) [reply src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs]: OK I'll change it.

File: src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs [hunk 2]
```diff
@@ -229,6 +400,12 @@ internal override void AsNativeHandle(IntPtr handle)
         /// <since_tizen> 3 </since_tizen>
         public int BitRate { get; }
 
+        /// <summary>
+        /// Gets the bit depth value of the current foramt.
```
Comment (m-macias) [inline src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs]:
```suggestion
        /// Gets the bit depth value of the current format.
```

Comment (hsgwon) [reply src/Tizen.Multimedia/MediaTool/AudioMediaFormat.cs]: I fixed it.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+        /// <summary>
+        /// A loudspeaker position located at center back of the middle layer.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        BackCenter,
+
+        /// <summary>
+        /// A Low Frequency Effects(band-limited low frequency channel) loudspeaker position located at the
+        /// bottom layer, and is normally at far right front of the bottom layer, when LFE1 is used.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: The enum is LFE2. Why did you explain LFE1?
Is LFE1 correct? If so, please fix it.

Suggestion:

/// A Low Frequency Effects(band-limited low frequency channel) loudspeaker is located in the bottom layer and is normally on the right front in the bottom layer when **LFE1** is used.  _(maybe LFE2)_

If you want to separate lines:

/// A Low Frequency Effects(band-limited low frequency channel) loudspeaker is located in the bottom layer
/// and is normally on the right front in the bottom layer when **LFE1** is used.  _(maybe LFE2)_

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+        /// <summary>
+        /// A Low Frequency Effects(band-limited low frequency channel) loudspeaker position located at the
+        /// bottom layer, and is normally at far right front of the bottom layer, when LFE1 is used. 
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        LFE2,
+
+        /// <summary>
+        /// A loudspeaker position located at left side of the middle layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located on the left side in the middle layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        SideLeft,
+
+        /// <summary>
+        /// A loudspeaker position located at right side of the middle layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located on the right side in the middle layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        SideRight,
+
+        /// <summary>
+        /// A loudspeaker position located at far left front of the top layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located at the left front in the top layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        TopFrontLeft,
+
+        /// <summary>
+        /// A loudspeaker position located at far right front of the top layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located on the right front in the top layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        TopFrontRight,
+
+        /// <summary>
+        /// A loudspeaker position located at center front of the top layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located at the center front in the top layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+        Mono = -2,
+
+        /// <summary>
+        /// Invalid position.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        Invaild = -1,
+
+        /// <summary>
+        /// A loudspeaker position located at far left and centered vertically with the middle layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located vertically at the left and center in the middle layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        FrontLeft = 0,
+
+        /// <summary>
+        /// A loudspeaker position located at far right and centered vertically with the middle layer.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located vertically at the right and center in the middle layer.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        FrontRight,
+
+        /// <summary>
+        /// A loudspeaker position located at the middle layer corresponding to the center of the television
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located in the middle layer surrounding the center of the television screen as viewed from the seating area.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+        /// screen as viewed from the seating area.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        FrontCenter,
+
+        /// <summary>
+        /// A Low Frequency Effects(band-limited low frequency channel) loudspeaker position located at
+        /// the bottom layer and normally far left front, when LFE2 is used.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: The enum is LFE1. Why did you explain LFE2?
Is LFE1 correct? If so, please fix it.

Suggestion:

/// A Low Frequency Effects(band-limited low frequency channel) loudspeaker is located in the bottom layer and is normally at the left front in the bottom layer when **LFE2** is used.  _(maybe LFE1)_

If you want to separate lines:

/// A Low Frequency Effects(band-limited low frequency channel) loudspeaker is located in the bottom layer
/// and is normally at the left front in the bottom layer when **LFE2** is used.  _(maybe LFE1)_

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+        TopFrontRight,
+
+        /// <summary>
+        /// A loudspeaker position located at center front of the top layer.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        TopFrontCenter,
+
+        /// <summary>
+        /// A loudspeaker position located at the center of the top layer directly above the seating area.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: A loudspeaker located in the middle layer surrounding the center of the television screen as viewed from the seating area.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+        WideRight,
+
+        /// <summary>
+        /// A loudspeaker position located between back left and side left.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        SurroundLeft,
+
+        /// <summary>
+        /// A loudspeaker position located between back right and side right.
```
Comment (journey2w) [inline src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: Please see the above comments.

File: src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+
+        /// <summary>
+        /// A Low Frequency Effects(band-limited low frequency channel) loudspeaker position located at
+        /// the bottom layer and normally far left front, when LFE2 is used.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        LFE1,
+
+        /// <summary>
+        /// A loudspeaker position located at far left back of the middle layer.
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia/MediaTool/MediaFormatChannelPosition.cs]: All statements described here is from SMPTE 2036-2-2008 standard document.
So, I think it's better to keep these statements.

**Pull request merged**